newthread.php

```diff
@@ -1125,18 +1125,7 @@
 		}
 	}
 
-	$php_max_upload_filesize = return_bytes(ini_get('max_upload_filesize'));
-	$php_post_max_size = return_bytes(ini_get('post_max_size'));
-
-	if ($php_max_upload_filesize != 0 && $php_post_max_size != 0)
-	{
-		$php_max_upload_size = min($php_max_upload_filesize, $php_post_max_size);
-	}
-	else
-	{
-		$php_max_upload_size = max($php_max_upload_filesize, $php_post_max_size);
-	}
-
+	$php_max_upload_size = min(array_filter(array(return_bytes(ini_get('upload_max_filesize')), return_bytes(ini_get('post_max_size')))));
```

Comment:
As this logic is spread across 3 places (at the minute - it might spread further with plugins and such), it would make a lot of sense to create a function to do this.

It might also be nice to split this into a few lines to make reading easier if it is moved to a function:

```suggestion
	$php_max_upload_size = @min(
		array_filter(
			array(
				return_bytes(ini_get('upload_max_filesize')), 
				return_bytes(ini_get('post_max_size'))
			)
		)
	);
```

Reply:
While multilining makes code3 more readable, I honestly think this is not needed here. MyBB is written in `php` and - I hope ) no dev has any trouble reading plain-php code.........